core/templates/dev/head/domain/collection/editable-collection-backend-api.service.spec.ts

```diff
@@ -15,140 +15,123 @@
-      return deferred.promise;
+  };
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule]
     });
 
-    // Sample collection object returnable from the backend
-    sampleDataResults = {
-      collection: {
-        id: '0',
-        title: 'Collection Under Test',
-        category: 'Test',
-        objective: 'To pass',
-        version: '1',
-        nodes: [{
-          exploration_id: '0'
-        }],
-        next_exploration_ids: [],
-        completed_exploration_ids: []
-      }
-    };
-  }));
-
-  afterEach(function() {
-    $httpBackend.verifyNoOutstandingExpectation();
-    $httpBackend.verifyNoOutstandingRequest();
+    editableCollectionBackendApiService = TestBed.get(
+      EditableCollectionBackendApiService);
+    httpTestingController = TestBed.get(HttpTestingController);
+  });
+
+  afterEach(() => {
+    httpTestingController.verify();
   });
 
   it('should successfully fetch an existing collection from the backend',
-    function() {
+    fakeAsync(() => {
       var successHandler = jasmine.createSpy('success');
       var failHandler = jasmine.createSpy('fail');
 
-      $httpBackend.expect('GET', '/collection_editor_handler/data/0').respond(
-        sampleDataResults);
-      EditableCollectionBackendApiService.fetchCollection('0').then(
+      editableCollectionBackendApiService.fetchCollection('0').then(
         successHandler, failHandler);
-      $httpBackend.flush();
+      var req = httpTestingController.expectOne(
+        '/collection_editor_handler/data/0');
+      expect(req.request.method).toEqual('GET');
+      req.flush(sampleDataResults);
+
+      flushMicrotasks();
 
       expect(successHandler).toHaveBeenCalledWith(sampleDataResults.collection);
       expect(failHandler).not.toHaveBeenCalled();
-    }
+    })
   );
 
   it('should use the rejection handler if the backend request failed',
-    function() {
+    fakeAsync(() => {
       var successHandler = jasmine.createSpy('success');
       var failHandler = jasmine.createSpy('fail');
 
-      // Loading a collection the first time should fetch it from the backend.
-      $httpBackend.expect('GET', '/collection_editor_handler/data/1').respond(
-        500, 'Error loading collection 1.');
-      EditableCollectionBackendApiService.fetchCollection('1').then(
+      editableCollectionBackendApiService.fetchCollection('1').then(
         successHandler, failHandler);
-      $httpBackend.flush();
+      var req = httpTestingController.expectOne(
+        '/collection_editor_handler/data/1');
+      expect(req.request.method).toEqual('GET');
+      req.flush('Error loading collection 1', {
+        status: 500, statusText: 'Invalid Request'
+      });
+
+      flushMicrotasks();
 
       expect(successHandler).not.toHaveBeenCalled();
-      expect(failHandler).toHaveBeenCalledWith('Error loading collection 1.');
-    }
+      expect(failHandler).toHaveBeenCalled();
```

Comment:
Can we expect the param like before (ie. use `toHaveBeenCalledWith`)?

Reply:
Done